### ISSUE TITLE
Type error during build

### ISSUE THREAD

**Issue by krzar** [sidorares/node-mysql2#1866]
While building my project I get an error:

```
node_modules/mysql2/index.d.ts:86:18 - error TS2320: Interface 'PoolConnection' cannot simultaneously extend types 'PoolConnection' and 'Connection'.
  Named property 'execute' of types 'PoolConnection' and 'Connection' are not identical.

86 export interface PoolConnection extends mysql.PoolConnection, Connection {
                    ~~~~~~~~~~~~~~

node_modules/mysql2/index.d.ts:86:18 - error TS2320: Interface 'PoolConnection' cannot simultaneously extend types 'PoolConnection' and 'Connection'.
  Named property 'unprepare' of types 'PoolConnection' and 'Connection' are not identical.

86 export interface PoolConnection extends mysql.PoolConnection, Connection {
```

The version of mysql2 I am using is `3.1.2`, I can build the project using `--skipLibCheck` but I would prefer to avoid it.

**Comment by wellwelwel** [comment]
> @krzar, sorry for the delay, this issue has been fixed in `3.2.2` version.

Can you check it again in `^3.4.1`?

**Comment by wellwelwel** [comment]
Hi @krzar, we currently have a **TypeScript Build Test** since the #1957 was merged.

Also, this issue has been fixed in version `^3.2.2` in PR #1955.

---

I'm closing this **Issue**, but feel free to ask anything 🙋🏻‍♂️
